docs/iam-usage.rst

```diff
@@ -0,0 +1,144 @@
+Using IAM For Users
+===================
+
+.. note:: This document does not explain how Google Cloud IAM works,
+   for that, please check out `the docs <https://cloud.google.com/iam/docs/>`_.
+
+Resources that implement the IAM interface provide the following methods:
+
+Policy Editing Methods
+----------------------
+
+.. note:: All policy editing methods except ``set_policy`` have an optional
+   ``retries`` keyword argument that takes the number of times to retry in the
+   even of an interrupted transaction
+
+>>> resource.add_roles(iam.user('alice@example.com'), iam.roles.OWNER.name, iam.roles.EDITOR.name)
+True
```

Comment:
I'm also trying to understand how this maps to the API. Is it fair to say that add_roles is a convenience method on top of `serviceAccounts.setIamPolicy` or is it something else?